library/webpack-loaders/web-worker-client-loader.js

```diff
@@ -0,0 +1,85 @@
+const { relative } = require('path')
+
+module.exports = WebworkerClientLoader
+
+function WebworkerClientLoader(source, map) {
+  const filename = relative(this.rootContext, this.resourcePath)
+  return createClientCode({ filename })
+}
+
+function createClientCode({ filename }) {
+  return `|export function useWebWorker() {
+          |  const workerRef = React.useRef(createProxy())
+          |  React.useEffect(
+          |    () => {
+          |      // create a new worker
+          |      const workerPath = __webpack_public_path__ +
+          |        __webpack_web_worker_chunk_manifest__[${JSON.stringify(filename)}].filename
```

Comment:
```suggestion
          |      export default __webpack_public_path__ +
          |        __webpack_web_worker_chunk_manifest__[${JSON.stringify(filename)}].filename
```

Reply:
Damn, ik ben vergeten te melden waardoor het komt (ik bedacht het me tijdens het douchen die avond). De `import` van de webworker gebeurt ook bij client side rendering en daar is die asset niet beschikbaar. Dat genereert de error.

Een aantal mogelijke oplossingen:

1. Zorgen dat we hetzelfde doen als de webpack 5 manier en met behulp van een parser hook zorgen dat we detecteren wanneer er een webworker script gebruikt wordt.
2. De code fool-proof maken en zorgen dat die een lege string of undefined terug geeft (eventueel met optional chaining)
3. Zorgen dat de code niet geïmporteerd wordt in de server side renderer door een soort van 'noop' loader te gebruiken (doen we ergens anders ook al)
4. In plaats van `import` `require` gebruiken op de plek waar je de url nodig hebt.

Die laatste is denk ik het makkelijkst: `new Worker(require('./my-worker.js?webworker'))`. Voor onze eigen gezondheid zou ik wel de foutmelding verbeteren, zoiets:

```js
if (!__webpack_web_worker_chunk_manifest__) throw new Error('Only import this file in a client-side webpack context')
```